vueapp/src/components/Game.vue

```diff
@@ -43,88 +47,79 @@
-                
-                axios
-                    .get(endpoint)
-                    .then(function(resp)
-                    { 
-                        response = resp;
-                        console.log("response: ", response);
-                        if (response.status == 204)
-                            found = false;
-                        else { 
-                            g.romanizedTitle = resp.data.romanizedTitle;
-                            g.gameTitle = resp.data.title;
-                            g.series = resp.data.series;
-                            g.sharedMechanics = resp.data.sharedMechanics;
-                            g.avgPlayLength = resp.data.averagePlayLength;
-                        }
-                    })
-                    .then(function(){
-                        //console.log("response 2: ", resp);
-                        if(!found){
-                            console.log("Searching stubs by name...");
-                            
-                            endpoint = URLs.LOCALROOT + URLs.DATASTUB + searchId;
-                            endpoint = endpoint.toLowerCase();
-                            
-                            axios 
-                                .get(endpoint)
-                                .then(function (resp){
-                                    response = resp;
-                                    console.log("response: ", response);
-                                    if (response.status == 204)
-                                        found = false;
-                                    else { 
-                                        g.romanizedTitle = resp.data.romanizedTitle;
-                                        g.gameTitle = resp.data.title;
-                                        g.series = resp.data.series;
-                                        g.sharedMechanics = resp.data.sharedMechanics;
-                                        g.avgPlayLength = resp.data.averagePlayLength;
-                                    }
-                                })
-                                .catch(function(err) { 
-                                    console.log("error:", err);
-                            });
-                            console.log("endpoint: ", endpoint);
-                            
-                        }
-                    })
-                    .catch(function(err)
-                    { 
-                        alert("error:", err); 
-                    })
-                    .finally(function(){
-                        console.log(g);
-                    })
-            
-            }
+                return axios.get(endpoint);
+            }            
         },
         created: function() {
-            this.loadCard();
+            let p = this.loadCard() // returns P<Game>
+                .then( async function(resp) {
+                    gcc.romanizedTitle = resp.data.romanizedTitle;
```

Comment:
~~If the property names are the same, can we just return `resp.data` instead of mapping it to a separate object? Or If not, could we do this in a constructor of the object?~~

Realized that this is a vue object. I'm wondering if we could create a method in the object itself to take this data in and set the properties. Just thinking that we could potentially have this same code in a few different places so it might make sense to put it somewhere central.

Reply:
There is no clean way to enumerate explicitly assigned properties over all of the other junk the promise object contains, so I went the easier hard coded route. Each object page will need its own block like this but at leasg it will also make it visually obvious what properties are enumerated

Reply:
Makes sense. I guess that's probably one advantage of a framework like angular has over vue where all that stuff is built in by default. This works!